### PR TITLE
PHP - Mail sample generation using promises generics

### PR DESCRIPTION
**Pull request by Ndiritu**

### CODE REVIEW

File: .github/workflows/get-started-php.yml [hunk 1]
```diff
@@ -8,11 +8,13 @@ on:
     paths:
       - 'get-started/azure-auth/php/**'
       - 'get-started/quickstart/php/**'
+      - 'msgraph-mail/php/**'
```
Comment (baywet) [inline .github/workflows/get-started-php.yml:11]: I'm not sure why we're adding this path?

File: msgraph-mail/php/composer.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "microsoft/kiota-samples",
   "require": {
-    "microsoft/kiota-abstractions": "^0.8.0",
-    "microsoft/kiota-http-guzzle": "^0.8.2",
-    "microsoft/kiota-serialization-text": "^0.7.0",
-    "microsoft/kiota-serialization-json": "^0.6.0",
-    "microsoft/kiota-authentication-phpleague": "^0.8.2"
+    "microsoft/kiota-abstractions": "dev-fix/promises-generics",
```
Comment (baywet) [inline msgraph-mail/php/composer.json:4]: this will eventually need to be reverted

Comment (Ndiritu) [reply msgraph-mail/php/composer.json]: Bumped to stable

File: .github/workflows/get-started-php.yml (hunk 1 above)
Comment (Ndiritu) [reply .github/workflows/get-started-php.yml]: Reverted.

**Review by baywet — APPROVED** [review]

**Pull request merged**